src/rpc_connection.c

```diff
@@ -603,13 +605,13 @@ on_rpc_abort(rpc_connection_t conn, rpc_object_t args __unused, rpc_object_t id)
 	g_rw_lock_reader_unlock(&conn->rco_icall_rwlock);
 	call->rc_ended = true;
 	call->rc_aborted = true;
+	rpc_connection_call_retain(call);
 	notify_signal(&call->rc_notify);
-	g_mutex_unlock(&call->rc_mtx);
-
-	if (call->rc_abort_handler) {
-		call->rc_abort_handler();
-		Block_release(call->rc_abort_handler);
-		call->rc_abort_handler = NULL;
+	if (call->rc_abort_handler)
+		call_abort_locked(call);
```

Comment:
mutex lock and unlock are not paired in the same function.

Reply:
call_abort_locked() has to drop the lock to call the abort function. After the abort function is called the call is released. And after that, the call can't be accessed. This can be changed to 1. take the lock again even if it isn't needed 2. return to the caller 3 make the caller drop the lock and release the call. That seems very inefficient and is more duplication.